client/extensions/woocommerce/app/settings/shipping/shipping-class-dialog.js

```diff
@@ -0,0 +1,218 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import { errorNotice } from 'state/notices/actions';
+import Dialog from 'components/dialog';
+import FormFieldSet from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+import FormClickToEditInput from 'woocommerce/components/form-click-to-edit-input';
+
+import {
+	getCurrentlyOpenShippingClass,
+	isCurrentlyOpenShippingClassNew,
+	getUsedShippingClassProps,
+} from 'woocommerce/state/ui/shipping/classes/selectors';
+import {
+	closeShippingClass,
+	updateShippingClassSetting,
+	saveCurrentlyOpenShippingClass,
+	removeShippingClass,
+} from 'woocommerce/state/ui/shipping/classes/actions';
+
+class ShippingClassDialog extends Component {
+	render() {
+		const { isNew, data, translate } = this.props;
+
+		if ( null === data ) {
+			return null;
+		}
+
+		// The modal can only be saved when there is a non-null name
+		const canSave = data.name && data.name.trim().length > 0;
+
+		const buttons = [
+			{
+				action: 'cancel',
+				label: translate( 'Cancel' ),
+			},
+			{
+				action: 'add',
+				label: isNew ? translate( 'Add' ) : translate( 'Done' ),
+				onClick: this.onSave,
+				isPrimary: true,
+				disabled: ! canSave,
+			},
+		];
+
+		if ( ! isNew ) {
+			buttons.unshift( {
+				action: 'delete',
+				label: (
+					<span>
+						<Gridicon icon="trash" /> { translate( 'Delete this class' ) }
+					</span>
+				),
+				onClick: this.onDelete,
+				additionalClassNames: 'shipping__class-delete is-scary is-borderless',
+			} );
+		}
+
+		const dialogProps = {
+			additionalClassNames: 'shipping__class-dialog woocommerce',
+			isVisible: true,
+			buttons: buttons,
+			onClose: this.onCancel,
+		};
+
+		return (
+			<Dialog { ...dialogProps }>
+				<div className="shipping__class-dialog-header">
+					<h3 className="shipping__class-dialog-heading">
+						{ isNew ? translate( 'Add shipping class' ) : translate( 'Edit shipping class' ) }
+					</h3>
+				</div>
+
+				<FormFieldSet className="shipping__class-dialog-name">
+					<FormLabel>{ translate( 'Name' ) }</FormLabel>
+					<FormTextInput type="text" name="name" value={ data.name } onChange={ this.onChange } />
+				</FormFieldSet>
+
+				<FormFieldSet className="shipping__class-dialog-slug">
+					<FormLabel>{ translate( 'Slug: ' ) }</FormLabel>
```

Comment:
Hi! I've found a possible matching string that has already been translated **95** times:
`translate( 'Slug' )` *ES Score: 17*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/26774)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).